core/go/internal/domainmgr/event_indexer.go

```diff
@@ -357,7 +357,7 @@ func (d *domain) handleEventBatchForContract(ctx context.Context, dbTX *gorm.DB,
 	}
 
 	// Then any finalizations of those states
-	if len(stateSpends) > 0 || len(stateReads) > 0 || len(stateConfirms) > 0 {
+	if len(stateSpends) > 0 || len(stateReads) > 0 || len(stateConfirms) > 0 || len(stateInfoRecords) > 0 {
```

Comment:
ok - this makes sense

So @awrichar this is saying that you can have a transaction that only includes `info`, and still results in a state receipt, as long as the info states are still emitted from the on-chain smart contract.

Does that fit your use case ok?

Reply:
Yes, exactly.

FWIW I don't think it's where I've actually ended up in my use case at the moment - I believe my transactions ended up having at least one other type of state, and not only info states. But fundamentally I don't see why we would _ignore_ the domain telling us to confirm info states, simply because other states weren't present.